src/main/java/com/yellow/ordermanageryellow/controller/ProductController.java

```diff
@@ -0,0 +1,70 @@
+package com.yellow.ordermanageryellow.controller;
+
+import com.yellow.ordermanageryellow.Exception.ObjectAllReadyExists;
+import com.yellow.ordermanageryellow.Service.ProductService;
+import com.yellow.ordermanageryellow.model.Product;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.NoSuchElementException;
+
+@RequestMapping("/product")
+@RestController
+public class ProductController {
+   @Autowired
+    private ProductService productService;
+
+    @PostMapping
+    public Product addProduct(@RequestBody Product product) throws ObjectAllReadyExists {
+        return  productService.addProduct(product);
+    }
+
+    @GetMapping("/names/{prefix}")
+    public ResponseEntity getAllNamesProducts(@PathVariable("prefix") String prefix,@RequestHeader ("token")String token){
+        try{
+        return ResponseEntity.status(HttpStatus.OK).body(productService.getAllProductsNames(token,prefix)) ;
+    }
+    catch (Exception e ){
+            return  ResponseEntity.status(HttpStatus.INTERNAL_SERVER_ERROR).body(e.getMessage()) ;
+        }
+    }
+    @GetMapping
+    public ResponseEntity getAllProduct(){
```

Comment:
you should retrieve products related to company

Reply:
Ayala Shprey will add the token  for all of us. That's why I didn't do these tests